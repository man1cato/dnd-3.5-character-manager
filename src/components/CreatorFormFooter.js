import React from 'react'

const CreatorFormFooter = ({
	page, 
	pages, 
	handleBack, 
	handleNext, 
	handleSubmit, 
	setErrors, 
	isSubmitting, 
	isValid
}) => (
	<div className="container container--footer form__footer">
		{page !== 1 && (
			<button type="button" key="backButton" onClick={() => handleBack(setErrors)}>   
				Back
			</button>
		)}
		{page === pages.length ? (
			<button type="submit" key="submitButton" disabled={isSubmitting || !isValid} onClick={handleSubmit}>   
				Create Character
			</button>
		) : (                                        
			<button type="button" key="nextButton" disabled={!isValid} onClick={handleNext}>   
				Next
			</button>
		)}
	</div>
)

export default CreatorFormFooter