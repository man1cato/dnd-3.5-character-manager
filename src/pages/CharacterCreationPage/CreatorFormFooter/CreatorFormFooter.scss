@import '../CharacterCreationPage.scss';


.CreatorFormFooter {
   position: fixed;
   bottom: 0px;
   height: 10vh;
   min-height: 60px;
   width: -webkit-fill-available;
   background: white;
   box-shadow: 0 0 5px $light-grey;
   padding: $s-size 0;
}

.CreatorFormFooter__content {
   display: flex;
   justify-content: space-between;
   > * {
      width: -webkit-fill-available;
   }
   :nth-child(2) {
      margin-left: $s-size;
   }
}