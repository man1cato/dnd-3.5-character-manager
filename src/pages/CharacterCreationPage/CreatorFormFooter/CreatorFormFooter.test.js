import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import CreatorFormFooter from './CreatorFormFooter'


const handleBack = jest.fn() 
const handleNext = jest.fn() 

const props = {
    page: 1,
    pages: ['Page1', 'Page2', 'Page3'],
    handleBack,
    handleNext,
    isSubmitting: false,
    isValid: true
}

test('should render CreatorFormFooter', () => {
    const { container } = render(<CreatorFormFooter {...props} />)
    expect(container.firstChild).toMatchSnapshot()
})