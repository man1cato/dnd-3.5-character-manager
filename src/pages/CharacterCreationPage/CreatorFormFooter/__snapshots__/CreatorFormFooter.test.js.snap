// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreatorFormFooter 1`] = `
<footer
  class="CreatorFormFooter"
>
  <div
    class="container CreatorFormFooter__content"
  >
    <button
      class="button--form-nav button--primary"
      id="nextButton"
      type="button"
    >
      Next
    </button>
  </div>
</footer>
`;
